proposal.md

```diff
@@ -0,0 +1,135 @@
+# Client-Server Split
+
+## What is this?
+
+This is an RFC, for a major change in design. I propose to split `deploy-rs` into two fully self standing executables, a
+server and a client. This would allow for additional flexibility and some issues are impossible to solve nicely with the
+current design.
+
+## The design
+
+I propose to split `deploy-rs` into a server and a client, `deploy` and `activate` respectively. Basically, while now
+`activate` is just a helper, this RFC proposes to make it a self standing program. The basic jist is that the two
+components would communicate over a open ssh connection, using plain REST (this is important later). That would allow us
```

Comment:
We'd have to make a new ssh connection to confirm activation anyways, since that's precisely what we're confirming (the ability to ssh into the machine).

Also, we'd need to reimplement `nix copy` for this. Should not be too difficult, but something to keep in mind.

Reply:
Well we'd still be down to 2 ssh connections from 3 or 4 or however many we have now